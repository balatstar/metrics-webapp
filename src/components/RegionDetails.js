// RegionDetails.js
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchRegions } from '../redux/region/regionSlice';
import './RegionList.css';

const RegionDetails = () => {
  const dispatch = useDispatch();
  const { regionId } = useParams();
  const regionIdAsNumber = parseInt(regionId, 10);

  useEffect(() => {
    dispatch(fetchRegions());
  }, [dispatch]);

  const regions = useSelector((state) => state.regions.regions);
  const loading = useSelector((state) => state.regions.loading);
  const error = useSelector((state) => state.regions.error);

  if (loading === 'loading') {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  const region = regions.find((r) => r.regionid === regionIdAsNumber);

  if (!region) {
    return <div>Region not found</div>;
  }

  return (
    <div>
      <div className="region-title">
        <div className="region-map"></div>
        <h2>{region.shortname}<br />
        <small>Forecast: {region.intensity.forecast}<br />{region.intensity.index}</small></h2>
      </div>
      <div className="region-head"><h3>Generation Mix</h3></div>
      <ul className="region-content">
        {region.generationmix.map((mix) => (
          <li className="region-content-links" key={mix.fuel}>
            {mix.fuel}: {mix.perc}%
          </li>
        ))}
      </ul>
  </div>
  );
};

export default RegionDetails;
