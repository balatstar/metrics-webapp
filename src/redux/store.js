import { configureStore, combineReducers } from '@reduxjs/toolkit';
import regionReducer from '../redux/region/regionSlice';

const rootReducer = combineReducers({
  regions: regionReducer,
});

const initialState = {
  regions: {
    regions: [],
    loading: 'idle',
    error: null,
    uniqueIntensityIndexValues: [],
    lowestForecastValues: {},
  },
};

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});

export default store;
