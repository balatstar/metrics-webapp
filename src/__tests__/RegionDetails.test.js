import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import RegionDetails from '../components/RegionDetails';

describe('RegionDetails', () => {
  it('renders RegionDetails component', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <RegionDetails />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
