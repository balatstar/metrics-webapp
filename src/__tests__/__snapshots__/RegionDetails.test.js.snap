// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionDetails renders RegionDetails component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
